src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs

```diff
@@ -51,7 +48,7 @@ internal sealed class SessionWithSolution : IDisposable
             }
         }
 
-        private SessionWithSolution(RemoteHostClient.Connection connection, PinnedRemotableDataScope scope)
+        private SessionWithSolution(ref OwnedDisposable<RemoteHostClient.Connection> connection, PinnedRemotableDataScope scope)
         {
             _connection = connection;
```

Comment:
shouldn't this has Move? otherwise, won't using (.. Box()) will dispose the connection?

Reply:
Yes, the instance was incorrectly copied by value. Corrected in 3075c31dc46a066a2cab590683f75711bda80a5f.